lib/pipelineTemplateVersionFactory.js

```diff
@@ -130,6 +130,47 @@ class PipelineTemplateVersionFactory extends BaseFactory {
         return newPipelineTemplateVersion;
     }
 
+    /**
+     * List all the versions of a pipeline template
+     * @method create
+     * @param  {Object}     config               Config object
+     * @param templateMetaFactory
+     * @param  {Datastore}  config.datastore     Object that will perform operations on the datastore
+     * @param  {String}     config.name          The template name
+     * @param  {String}     config.namespace     The template namespace
+     * @param  {String}     config.templateId    Identifier of the template
+     * @param  {Object}     config.paginate         Pagination parameters
+     * @param  {Number}     config.paginate.count   Number of items per page
+     * @param  {Number}     config.paginate.page    Specific page of the set to return
+     * @return {Promise}                         List of PipelineTemplateVersionModel
+     */
+    async list(config, templateMetaFactory) {
+        const { name, namespace, templateId } = config;
+
+        const params = {};
+
+        if (templateId) {
+            params.id = templateId;
+        }
+        if (name && namespace) {
+            params.name = name;
+            params.namespace = namespace;
+        }
+
+        const pipelineTemplateMeta = await templateMetaFactory.get(params);
+
+        if (!pipelineTemplateMeta) {
+            return [];
+        }
+
+        return super.list({
+            params: {
+                templateId: pipelineTemplateMeta.id
+            },
+            paginate: config.paginate
+        });
+    }
```

Comment:
The list API should be generic to allow filtering on any field. Accept filter criteria `config.params` and pass it with any modifications to the query.

```suggestion
        return super.list(config);
    }
```